src/org/labkey/test/tests/list/ListDateAndTimeTest.java

```diff
@@ -0,0 +1,1434 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+@Category({Daily.class, Data.class, Hosting.class})
+public class ListDateAndTimeTest extends BaseWebDriverTest
+{
+
+    private static final String PROJECT_NAME = "List Date And Time Test";
+    @Override
+    public List<String> getAssociatedModules()
+    {
+        return Arrays.asList("list");
+    }
+
+    @Override
+    protected String getProjectName()
+    {
+        return PROJECT_NAME;
+    }
+
+    @BeforeClass
+    public static void setupProject()
+    {
+        ListDateAndTimeTest init = (ListDateAndTimeTest)getCurrentTest();
+        init.doSetup();
+    }
+
+    private void doSetup()
+    {
+        log("Setup project and list module");
+        _containerHelper.createProject(PROJECT_NAME, null);
+        goToProjectHome();
+    }
+
+    @Override
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        _containerHelper.deleteProject(getProjectName(), afterTest);
+    }
+
+    @AfterClass
+    public static void resetAfterClass()
+    {
+        ListDateAndTimeTest init = (ListDateAndTimeTest) getCurrentTest();
+        init.resetSiteSettings();
+    }
+
+    public void resetSiteSettings()
+    {
+        log("Reset site settings.");
+        LookAndFeelSettingsPage.beginAt(this).reset();
+    }
+
+    @Before
+    public void preTest()
+    {
+        goToProjectHome();
+        if (isElementPresent(PortalHelper.Locators.webPartTitle("Search")))
+            new PortalHelper(this).removeWebPart("Search");
+    }
+
+    @Override
+    protected BrowserType bestBrowser()
+    {
+        return BrowserType.CHROME;
+    }
+
+    private SimpleDateFormat _defaultDateFormat = new SimpleDateFormat("yyyy-MM-dd");
+    private SimpleDateFormat _defaultTimeFormat = new SimpleDateFormat("HH:mm:ss");
+    private SimpleDateFormat _defaultDateTimeFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+
+    private void prepForDateAndTimeOnlyTest(String listName) throws IOException, CommandException
+    {
+        log("Reset site setting to have default formats for date and time values.");
+        LookAndFeelSettingsPage settingsPage = LookAndFeelSettingsPage.beginAt(this);
+        settingsPage.reset();
+
+        // Reset doesn't wait for page (page may or may not refresh).
+        sleep(500);
+
+        // Use java's Date object and SimpleDateFormatter (not strings) to enter the data.
+        _defaultDateFormat = new SimpleDateFormat(settingsPage.getDefaultDateDisplay());
+        _defaultTimeFormat = new SimpleDateFormat(settingsPage.getDefaultTimeDisplay());
+        _defaultDateTimeFormat = new SimpleDateFormat(settingsPage.getDefaultDateTimeDisplay());
+
+        deleteListIfPresent(getProjectName(), listName);
+
+    }
```

Comment:
This does a lot of unnecessary UI groveling. Only getting the default formats requires going through the UI and that only needs to happen once (during `doSetup` would be good). The rest of this (resetting the look and feel and deleting the list) can be done via API.